[#7675][feat] CapturedGraph to support max_batch_size > max(cuda_graph_batch_sizes)

MrGeva opened the pull request:
A recent change to the CapturedGraph captured cuda graphs only up to the highest value of cuda_graph_batch_sizes.
so when the user set a higher max_batch_size it will fail on assertion in CapturedGraph because input batch size dim will be higher than the maximal cuda_graph_batch_sizes value. 

In this change, we capture up to the max value of cuda_graph_batch_sizes and fallback to non-captured graph when higher batch size occurs.

renamed CapturedGraph's max_batch_size to cuda_graph_batch_sizes_max

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Dynamic batching for CUDA graph replay with rounded batch sizes.
  - Automatic clamping, deduplication, and sorting of provided CUDA graph batch sizes.
  - Heuristic generation of batch sizes when none are provided.

- Improvements
  - Separate maximum for CUDA graph batch sizes with clearer validation and warm-up logging.
  - Added debug logs for clamping and fallback scenarios.

- Behavior Changes
  - Graceful fallback to regular execution when batches can’t be rounded or exceed limits.
  - Renamed a related public attribute for clarity.

- Tests
  - Comprehensive unit tests for clamping, heuristics, capture/replay, and fallbacks.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

<!--
Please write the PR title by following this template:

**[JIRA ticket/NVBugs ID/GitHub issue/None][type] Summary**

Valid ticket formats:
  - JIRA ticket: [TRTLLM-1234] or [FOOBAR-123] for other FOOBAR project
  - NVBugs ID: [https://nvbugs/1234567]
  - GitHub issue: [#1234]
  - No ticket: [None]

Valid types (lowercase): [fix], [feat], [doc], [infra], [chore], etc.

Examples:
  - [TRTLLM-1234][feat] Add new feature
  - [https://nvbugs/1234567][fix] Fix some bugs
  - [#1234][doc] Update documentation
  - [None][chore] Minor clean-up

Alternative (faster) way using CodeRabbit AI:

**[JIRA ticket/NVBugs ID/GitHub issue/None] @coderabbitai title**

NOTE: "@coderabbitai title" will be replaced by the title generated by CodeRabbit AI, that includes the "[type]" and title.
For more info, see /.coderabbit.yaml.

-->

## Description

<!--
Please explain the issue and the solution in short.
-->

## Test Coverage

<!--
Please list clearly what are the relevant test(s) that can safeguard the changes in the PR. This helps us to ensure we have sufficient test coverage for the PR.
-->

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- The reviewers assigned automatically/manually are appropriate for the PR.


- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--reuse-test (optional)pipeline-id --disable-fail-fast --skip-test --stage-list "A10-PyTorch-1, xxx" --gpu-type "A30, H100_PCIe" --test-backend "pytorch, cpp" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx" --detailed-log --debug(experimental)]`

Launch build/test pipelines. All previously running jobs will be killed.

`--reuse-test (optional)pipeline-id ` *(OPTIONAL)* : Allow the new pipeline to reuse build artifacts and skip successful test stages from a specified pipeline or the last pipeline if no pipeline-id is indicated. If the Git commit ID has changed, this option will be always ignored. The DEFAULT behavior of the bot is to reuse build artifacts and successful test results from the last pipeline.

`--disable-reuse-test ` *(OPTIONAL)* : Explicitly prevent the pipeline from reusing build artifacts and skipping successful test stages from a previous pipeline. Ensure that all builds and tests are run regardless of previous successes.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-PyTorch-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-PyTorch-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--test-backend "pytorch, cpp"` *(OPTIONAL)* : Skip test stages which don't match the specified backends. Only support [pytorch, cpp, tensorrt, triton]. Examples: "pytorch, cpp" (does not run test stages with tensorrt or triton backend). Note: Does **NOT** update GitHub pipeline status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests in addition to running L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-TensorRT-Post-Merge-1, xxx".

`--detailed-log ` *(OPTIONAL)* : Enable flushing out all logs to the Jenkins console. This will significantly increase the log volume and may slow down the job.

`--debug ` *(OPTIONAL)* : **Experimental feature**. Enable access to the CI container for debugging purpose. Note: Specify exactly one stage in the `stage-list` parameter to access the appropriate container environment. Note: Does **NOT** update GitHub check status.

For guidance on mapping tests to stage names, see `docs/source/reference/ci-overview.md`
and the `scripts/test_to_stage_mapping.py` helper.

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

yiqingy0 committed:
[None][infra] Bump version to 1.1.0rc5 (#7668)

Signed-off-by: Yiqing Yan <yiqingy@nvidia.com>

bo-nv committed:
[TRTLLM-7871][infra] Extend test_perf.py to add disagg-serving perf tests. (#7503)

Signed-off-by: Bo Deng <deemod@nvidia.com>

xinhe-nv committed:
[https://nvbugs/5494698][fix] skip gemma3 27b on blackwell (#7505)

Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

Linda-Stadter committed:
[https://nvbugs/5477359][fix] Nanobind: Allow none types for fields in result (#7672)

Signed-off-by: Linda-Stadter <57756729+Linda-Stadter@users.noreply.github.com>

leslie-fang25 committed:
[None][chore] remove executor config in kv cache creator (#7526)

Signed-off-by: leslie-fang25 <leslief@nvidia.com>

MrGeva committed:
fixed CapturedGraph max bs handling

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #18371](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/18371/) [ run ] triggered by Bot

MrGeva committed:
added copyright

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

nvamyt committed:
[https://nvbugs/5488212][waive] Waive failed tests for L20 (#7664)

Signed-off-by: nvamyt <amyt@nvidia.com>

tensorrt-cicd commented:
[PR_Github #18371](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/18371/) [ run ] completed with state `ABORTED`
[/LLM/main/L0_MergeRequest_PR pipeline #13779](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/13779/) completed with status: 'FAILURE'

v-shobhit committed:
[None][feat] Use a shell context to install dependancies (#7383)

Signed-off-by: Shobhit Verma <shobhitv@nvidia.com>
Signed-off-by: v-shobhit <161510941+v-shobhit@users.noreply.github.com>
Co-authored-by: Zhihan Jiang <68881590+nvzhihanj@users.noreply.github.com>

DomBrown committed:
[https://nvbugs/5505402] [fix] Disable deep_gemm for Qwen3 QKNormRoPEAttention and Linear layers due to accuracy issues (#7616)

Signed-off-by: Dom Brown <3886319+DomBrown@users.noreply.github.com>

EmmaQiaoCh committed:
[None][infra] Waive failed cases on main 0910 (#7676)

Signed-off-by: qqiao <qqiao@nvidia.com>
Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>
Co-authored-by: Yanchao Lu <yanchaol@nvidia.com>

MrGeva commented:
/bot run

suyoggupta commented on tensorrt_llm/_torch/auto_deploy/compile/backends/torch_cudagraph.py:
```diff
@@ -25,8 +25,8 @@ def __init__(
         self.model = model
-        self.max_batch_size = max(cuda_graph_batch_sizes)
-        ad_logger.info(f"Setting max batch size to {self.max_batch_size}")
+        self.cuda_graph_batch_sizes_max = max(cuda_graph_batch_sizes)
```
nit: cuda_graph_max_batch_size (to keep the naming consistent)

suyoggupta commented on tensorrt_llm/_torch/auto_deploy/compile/backends/torch_cudagraph.py:
```diff
@@ -86,22 +86,32 @@ def capture_graph(self, *args, **kwargs):
-        msg_bs = "Max batch size too small."
+        msg_bs = (
+            f"Input batch size exceeds maximum CUDA graph batch size. "
+            f"Max CUDA graph batch size: {self.cuda_graph_batch_sizes_max}, "
```
ditto

karljang committed:
[None][infra] Adjust labeling llm prompt for bug issues (#7385)

Signed-off-by: Kanghwan Jang <861393+karljang@users.noreply.github.com>

QiJune committed:
[None][ci] move some test cases from l40s to a30 (#7684)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

Fan-Yunfan committed:
[None][fix] Fix the incorrect header file import in dataType.h (#7133)

Signed-off-by: fanyunfan <2569548856@qq.com>
Co-authored-by: fanyunfan <2569658856@qq.com>
Co-authored-by: Yunfan Fan <46273019+fyf2016@users.noreply.github.com>
Co-authored-by: Kanghwan <861393+karljang@users.noreply.github.com>

chang-l committed:
[https://nvbugs/5498165][fix] fix permission error for config file lock (#7656)

Signed-off-by: Chang Liu <9713593+chang-l@users.noreply.github.com>

nv-guomingz committed:
[https://nvbugs/5513192][fix] Add the missing param for kv_cache_tran… (#7679)

Signed-off-by: nv-guomingz <137257613+nv-guomingz@users.noreply.github.com>

LinPoly committed:
[TRTLLM-1302][feat] Topk logprobs for TRT backend and top1 logprob for PyT backend (#6097)

Signed-off-by: Pengyun Lin <81065165+LinPoly@users.noreply.github.com>

ZhanruiSunCh committed:
[TRTLLM-7169][infra] Fix Slurm multi-node test showing "Submit Test Results" in the test name (#6856)

Signed-off-by: ZhanruiSunCh <184402041+ZhanruiSunCh@users.noreply.github.com>

ZhanruiSunCh committed:
[TRTLLM-6791][infra] Add check for uploading stage name and avoid overriding test result tar file (#6742)

Signed-off-by: ZhanruiSunCh <184402041+ZhanruiSunCh@users.noreply.github.com>
Signed-off-by: Zhanrui Sun <184402041+ZhanruiSunCh@users.noreply.github.com>
Co-authored-by: Yanchao Lu <yanchaol@nvidia.com>

chzblych committed:
[None][ci] Some improvements for Slurm CI (#7689)

Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>

chzblych committed:
[None][ci] Test waives for the main branch 09/14 (#7698)

Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>

PerkzZheng committed:
[None][feat] support gpt-oss with fp8 kv cache (#7612)

Signed-off-by: Perkz Zheng <67892460+PerkzZheng@users.noreply.github.com>

chang-l committed:
[TRTLLM-6903][feat] Support chunked prefill for multimodal models (#6843)

Signed-off-by: Chang Liu (Enterprise Products) <9713593+chang-l@users.noreply.github.com>

xinhe-nv committed:
[None][chore] Add failed cases into waives.txt (#7682)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>

JunyiXu-nv committed:
[None][chore] Enable multiple postprocess workers tests for chat completions api (#7602)

Signed-off-by: Junyi Xu <219237550+JunyiXu-nv@users.noreply.github.com>

crazydemo committed:
[TRTLLM-7279][test] add accuracy test for deepseek-r1 with chunked_prefill (#7365)

Signed-off-by: Ivy Zhang <25222398+crazydemo@users.noreply.github.com>

DylanChen-NV committed:
[https://nvbugs/5467981][fix] Fix Qwen2.5-VL fails with cuda graph padding (#7122)

Signed-off-by: Dylan Chen <191843203+DylanChen-NV@users.noreply.github.com>

HuiGao-NV committed:
[None][chore] move some cases from post-merge to pre-merge to detect errors in early stage (#7699)

Signed-off-by: Hui Gao <huig@nvidia.com>

Wanli-Jiang committed:
[TRTLLM-7918][feat] Support kvcache reuse for phi4mm (#7563)

Signed-off-by: Wanli Jiang <35160485+Wanli-Jiang@users.noreply.github.com>

ixlmar committed:
[None][test] add test for min_tokens (#7678)

Signed-off-by: ixlmar <206748156+ixlmar@users.noreply.github.com>

Wanli-Jiang committed:
[TRTLLM-7918][feat] Revert "Support kvcache reuse for phi4mm (#7563)" (#7722)

Signed-off-by: Wanli Jiang <35160485+Wanli-Jiang@users.noreply.github.com>

zhengd-nv committed:
[None][fix] using arrival time in llmapi when creating LlmRequest in pytorch workflow (#7553)

Signed-off-by: zhengd-nv <200704041+zhengd-nv@users.noreply.github.com>

jmydurant committed:
[TRTLLM-7192][feat] optimize MLA chunked prefill && support fp8 mla chunked prefill (#7477)

Signed-off-by: Mingyang Jiang <13463932+jmydurant@users.noreply.github.com>

chzblych committed:
[None][ci] Test waives for the main branch 09/15 (#7709)

Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>

MrGeva committed:
removed assert, renamed var, fixed test to cover bs > cud max bs

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

MrGeva committed:
capture first cuda graph on cuda_graph_max_batch_size

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #18653](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/18653/) [ run ] triggered by Bot

Commit:
[None][feat] Eagle, use last hidden post norm (#7546)

Signed-off-by: Izzy Putterman <iputterman@nvidia.com>

lucaslie committed:
[None][infra] AutoDeploy: codeowners for autodeploy unit tests (#7743)

Signed-off-by: Lucas Liebenwein <11156568+lucaslie@users.noreply.github.com>

ziyixiong-nv committed:
[TRTLLM-6668][feat] Enable overlap scheduler for two-model spec decoding (#7651)

Signed-off-by: ziyixiong-nv <219238287+ziyixiong-nv@users.noreply.github.com>

QiJune committed:
[None][ci] move qwen3 tests from GB200 to B200 (#7733)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

Nekofish-L committed:
[None][feat] support attention dp for qwen3 dense model (#7618)

Signed-off-by: Nekofish-L <liuxiangyang@mail.ustc.edu.cn>

Shixiaowei02 committed:
[None][doc] Fix the link in the doc (#7713)

Signed-off-by: Shi Xiaowei <39303645+Shixiaowei02@users.noreply.github.com>

VALLIS-NERIA committed:
[TRTLLM-4629] [feat] Add support of CUDA13 and sm103 devices (#7568)

Signed-off-by: Xiwen Yu <13230610+VALLIS-NERIA@users.noreply.github.com>
Signed-off-by: Tian Zheng <29906817+Tom-Zheng@users.noreply.github.com>
Signed-off-by: Daniel Stokes <dastokes@nvidia.com>
Signed-off-by: Zhanrui Sun <zhanruis@nvidia.com>
Signed-off-by: Xiwen Yu <xiweny@nvidia.com>
Signed-off-by: Jiagan Cheng <jiaganc@nvidia.com>
Signed-off-by: Yiqing Yan <yiqingy@nvidia.com>
Signed-off-by: Bo Deng <deemod@nvidia.com>
Signed-off-by: ZhanruiSunCh <184402041+ZhanruiSunCh@users.noreply.github.com>
Signed-off-by: xiweny <13230610+VALLIS-NERIA@users.noreply.github.com>
Co-authored-by: Tian Zheng <29906817+Tom-Zheng@users.noreply.github.com>
Co-authored-by: Daniel Stokes <dastokes@nvidia.com>
Co-authored-by: Zhanrui Sun <zhanruis@nvidia.com>
Co-authored-by: Jiagan Cheng <jiaganc@nvidia.com>
Co-authored-by: Yiqing Yan <yiqingy@nvidia.com>
Co-authored-by: Bo Deng <deemod@nvidia.com>
Co-authored-by: Zhanrui Sun <184402041+ZhanruiSunCh@users.noreply.github.com>

chzblych committed:
[TRTLLM-6295][test] Exit as early as possible and propagate exit status correctly for multi-node testing (#7739)

Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>

xinhe-nv committed:
[None][chore] Add failed cases into waives.txt (#7735)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>
Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

yilin-void committed:
[None][fix] Ensure that the W4A8 custom input scale remains aligned across all ranks (#7614)

Signed-off-by: Yilin Zhang <18275976+yilin-void@users.noreply.github.com>

bobboli committed:
[None][chore] Fix error when running trtllm-bench without cuda graph. (#7725)

Signed-off-by: Bo Li <22713281+bobboli@users.noreply.github.com>

nv-guomingz committed:
[None][doc] Clean the doc folder and move the outdated docs into lega… (#7729)

Signed-off-by: nv-guomingz <137257613+nv-guomingz@users.noreply.github.com>

limin2021 committed:
[TRTLLM-6898][feat] Add Cute DSL nvfp4 linear op (#7632)

Signed-off-by: Mindy Li <11663212+limin2021@users.noreply.github.com>

kaiyux committed:
[None] [chore] cherry pick changes on slurm scripts from `release/1.1.0rc2` (#7750)

Signed-off-by: Kaiyu Xie <26294424+kaiyux@users.noreply.github.com>

amitz-nv committed:
[https://nvbugs/5503529][fix] Change test_llmapi_example_multilora to get adapters path from cmd line to avoid downloading from HF (#7740)

Signed-off-by: Amit Zuker <203509407+amitz-nv@users.noreply.github.com>

xinhe-nv committed:
[TRTLLM-7070][feat] add gpt-oss serve benchmark tests (#7638)

Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

xinhe-nv committed:
[None][fix] waive hang tests on main (#7720)

Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

ziyixiong-nv committed:
[https://nvbugs/5471106][fix] Remove the waivers (#7711)

Signed-off-by: ziyixiong-nv <219238287+ziyixiong-nv@users.noreply.github.com>

xinhe-nv committed:
[None][chore] Add failed cases into waives.txt (#7746)

Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

byshiue committed:
Revert "[None][feat] support attention dp for qwen3 dense model" (#7765)

Tabrizian committed:
[TRTLLM-8044][refactor] Rename data -> cache for cacheTransceiver (#7659)

lucaslie reviewed (changes requested):
please run `trtllm-bench` before merging this PR

lucaslie commented on tensorrt_llm/_torch/auto_deploy/compile/backends/torch_cudagraph.py:
```diff
@@ -132,9 +142,24 @@ def forward(self, *args, **kwargs) -> Any:
         # Calculate rounded-up shapes for each input
+        # If any batch size exceeds max cuda_graph_batch_sizes, fall back to regular forward
+        rounded_batch_sizes = [
+            self.round_to_cuda_batch_size(input.shape[0]) for input in args_batched
+        ]
+
+        # If any rounded batch size is None (exceeds max), use regular forward
+        if any(bs is None for bs in rounded_batch_sizes):
+            actual_batch_sizes = [input.shape[0] for input in args_batched]
+            max_cuda_bs = max(self.cuda_graph_batch_sizes) if self.cuda_graph_batch_sizes else 0
+            ad_logger.debug(
+                f"Batch size {actual_batch_sizes} exceeds max CUDA graph batch size {max_cuda_bs}. "
+                f"Falling back to regular forward pass."
+            )
+            return self.model(*args, **kwargs)
+
         rounded_shapes = [
-            (self.round_to_cuda_batch_size(input.shape[0]),) + input.shape[1:]
-            for input in args_batched
+            (rounded_bs,) + input.shape[1:]
+            for rounded_bs, input in zip(rounded_batch_sizes, args_batched)
```
why is this change needed? the check below (`if combined_shape not in self.graphs`) can perfectly take care of this?

MrGeva replied:
I pulled the rounding to bs up to line 146 to be able to detect the case of bs > cuda_graph_max_batch_size, and execute eager in such case (line 158).

lucaslie committed:
[None][chore] AutoDeploy: neat disablement of transforms in pipeline (#7736)

Signed-off-by: Lucas Liebenwein <11156568+lucaslie@users.noreply.github.com>

achartier committed:
[None][chore] Remove unused get_quant_scales methods (#7687)

Signed-off-by: Aurelien Chartier <2567591+achartier@users.noreply.github.com>

yuanjingx87 committed:
[None][infra] add nspect allow list for false positive secrets (#5797)

Signed-off-by: Yuanjing Xue <197832395+yuanjingx87@users.noreply.github.com>

chang-l committed:
[TRTLLM-7398][doc] Add doc for KV cache salting support (#7772)

Signed-off-by: Chang Liu (Enterprise Products) <9713593+chang-l@users.noreply.github.com>

yuanjingx87 committed:
[None][infra] Update CI allowlist 2025-09-16 (#7773)

Signed-off-by: Yuanjing Xue <197832395+yuanjingx87@users.noreply.github.com>

yuanjingx87 committed:
[None][infra] Add nightly pipeline to generate lock files (#5798)

Signed-off-by: Yuanjing Xue <197832395+yuanjingx87@users.noreply.github.com>

HuiGao-NV committed:
[https://nvbugs/5516666][fix] cherrypick fix to the CUDA graph warmup issue when using speculative decoding (#7737)

Signed-off-by: Fanrong Li <23290157+lfr-0531@users.noreply.github.com>
Co-authored-by: Fanrong Li <23290157+lfr-0531@users.noreply.github.com>
Co-authored-by: Tao Li @ NVIDIA <tali@nvidia.com>
Co-authored-by: Signed-off-by: Hui Gao <huig@nvidia.com>

Tabrizian committed:
[None][waive] Waive tests (#7775)

Signed-off-by: Iman Tabrizian <10105175+Tabrizian@users.noreply.github.com>

timlee0212 committed:
[https://nvbugs/5489015][fix] Support communicator split in MNNVL allreduce and fix the binding issues. (#7387)

Signed-off-by: Shiyu Li <shili@nvidia.com>

hyukn committed:
[https://nvbugs/5488582][fix] Cherry-pick 7495: Avoid unexpected Triton recompilation in DG fused_moe (#7708)

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

kaiyux committed:
[TRTLLM-6741] [feat] enable LM tp for MTP, under attention dp case (cherry-pick #7128) (#7571)

Signed-off-by: Cheng Hang <chang@nvidia.com>
Co-authored-by: Cheng Hang <chang@nvidia.com>

lucaslie committed:
[None][chore] AutoDeploy: clean up of model unit test configuration (#7742)

Signed-off-by: Lucas Liebenwein <11156568+lucaslie@users.noreply.github.com>

QiJune committed:
[None][ci] waive test_llm_gemma_1gpu_summary_vswa (#7781)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

dc3671 committed:
[https://nvbugs/5517260][fix] move scaffolding contrib module's import to subdirectory (#7758)

Signed-off-by: Zhenhuan Chen <chenzhh3671@gmail.com>

QiJune committed:
[None][feat] add an example of KV cache host offloading (#7767)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

lfr-0531 committed:
[https://nvbugs/5485325][fix] Cherry-pick #7373: fix the CUDA graph warmup issue when using speculative decoding (#7734)

Signed-off-by: Fanrong Li <23290157+lfr-0531@users.noreply.github.com>
Co-authored-by: Tao Li @ NVIDIA <tali@nvidia.com>

QiJune committed:
[None][ci] waive test_llama_eagle3[True-FLASHINFER-False-False-False-False-True] (#7788)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

xinhe-nv committed:
[None][chore] Remove closed bugs (#7697)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>

ruodil committed:
[None][test] add gpt oss model for trtllm perf test (#7328)

Signed-off-by: Ruodi Lu <ruodil@nvidia.com>
Signed-off-by: Ruodi Lu <ruodil@users.noreply.github.com>
Co-authored-by: Ruodi Lu <ruodil@users.noreply.github.com>
Co-authored-by: Larry <197874197+LarryXFly@users.noreply.github.com>

xinhe-nv committed:
[TRTLLM-7250][fix] waive block tests (#7782)

Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

QiJune committed:
[None][doc] fix section header of llm_kv_cache_offloading example (#7795)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

2ez4bz committed:
[TRTLLM-7410][feat] Enable KV cache reuse and chunked prefill for mistral3.1 (#7628)

Signed-off-by: William Zhang <133824995+2ez4bz@users.noreply.github.com>

EmmaQiaoCh committed:
[None][infra] Waive failed tests on main (#7812)

Signed-off-by: qqiao <qqiao@nvidia.com>

asrivas committed:
[None][doc] Update Documentation link to point to docs instead of docs source code (#6495)

Signed-off-by: Anu <asrivastava274@gmail.com>

MatthiasKohl committed:
[TRTLLM-5966][feat] Helix: make softmax stats pointer available to attention gen (#6865)

Signed-off-by: Matthias Jouanneaux <mjoux@nvidia.com>
Co-authored-by: brb-nv <169953907+brb-nv@users.noreply.github.com>

yunruis committed:
[https://nvbugs/5516661][fix] Drop waive case 5516661 (#7791)

Signed-off-by: yunruis <205571022+yunruis@users.noreply.github.com>

netanel-haber committed:
[https://nvbugs/5508536][fix] Revert #7041: Move stop_criteria to sample_async (#7041) (#7796)

Signed-off-by: Netanel Haber <nhaber@nvidia.com>
Signed-off-by: Netanel Haber <58652339+netanel-haber@users.noreply.github.com>
Co-authored-by: Mike Iovine <miovine@nvidia.com>

lucaslie committed:
[#7308] [feat] AutoDeploy: graph-less transformers mode for HF (#7635)

Signed-off-by: h-guo18 <67671475+h-guo18@users.noreply.github.com>
Signed-off-by: Lucas Liebenwein <11156568+lucaslie@users.noreply.github.com>
Co-authored-by: h-guo18 <67671475+h-guo18@users.noreply.github.com>

Superjomn committed:
[None][ci] restore unwaive list (#7802)

Signed-off-by: Yan Chunwei <328693+Superjomn@users.noreply.github.com>

hyukn committed:
[None][fix] Make tile_tokens_dim calculation just in time before kernel launching. (#7529)

tile_tokens_dim directly depends on the num_token, which is a dynamic shape during tuning and inference. When AutoTuner prepares dummy tensors with different num_tokens, it does not update the value of tile_tokens_dim automatically. Therefore, the value stored in the AutoTuner cache is misaligned, which will introduce a lot of cache misses during inference, which hurts perf a lot.

To avoid this issue, we move the calculation of tile_tokens_dim right before kernel launching, so that the value of tile_tokens_dim is always up to date with the num_tokens of the current input tensor used for the kernel runner.

Also, the tile_tokens_dim is calculated based on the number of tokens of a tuned bucket, instead of the original token number. Because we only tune the value for the buckets, not for the raw input token number, to avoid unexpected misalignment between tile_tokens_dim and the token number.

This PR also removes the warmup requests with the extra input shapes, which are triggered in the CUDA graph warmup phase.

Signed-off-by: Yukun He <23156053+hyukn@users.noreply.github.com>

chzblych committed:
[None][chore] Version bump for 1.1.0rc6 (#7824)

Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>

nv-guomingz committed:
[https://nvbugs/5519544][fix] fix invalid expression for disabling pa… (#7806)

Signed-off-by: nv-guomingz <137257613+nv-guomingz@users.noreply.github.com>

crazydemo committed:
[TRTLLM-8070][test] add generation logits case for llama3 (#7759)

Signed-off-by: Ivy Zhang <25222398+crazydemo@users.noreply.github.com>

ziyixiong-nv committed:
[https://nvbugs/5523080][fix] Correct the batch index in device tensors (#7803)

Signed-off-by: ziyixiong-nv <219238287+ziyixiong-nv@users.noreply.github.com>

Barry-Delaney committed:
[None][feat] Cherry-pick DeepGEMM related commits from release/1.1.0rc2 (#7716)

Signed-off-by: Barry Kang <43644113+Barry-Delaney@users.noreply.github.com>

limin2021 committed:
[None][fix] Fix CI issue for dsl pkg install (#7784)

Signed-off-by: Mindy Li <11663212+limin2021@users.noreply.github.com>
Co-authored-by: QI JUN <22017000+QiJune@users.noreply.github.com>

ixlmar committed:
[https://nvbugs/5508890][fix] gen. result cleanup when using PostprocWorker (#7771)

Signed-off-by: ixlmar <206748156+ixlmar@users.noreply.github.com>

yuanjingx87 committed:
[None][infra] update ci allow list 2025/09/17 (#7816)

Signed-off-by: Yuanjing Xue <197832395+yuanjingx87@users.noreply.github.com>

leslie-fang25 committed:
[None][chore] Remove executor config in create_py_executor (#7599)

Signed-off-by: leslie-fang25 <leslief@nvidia.com>

xinhe-nv committed:
[None][chore] Add failed cases into waives.txt (#7801)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>

dongfengy committed:
[https://nvbugs/5519530][fix] Fix gptoss 2-gpu test (#7819)

Signed-off-by: Dongfeng Yu <dongfengy@nvidia.com>

Wanli-Jiang committed:
[TRTLLM-6577][feat] Support nano_v2_vlm in pytorch backend (#7207)

Signed-off-by: Wanli Jiang <35160485+Wanli-Jiang@users.noreply.github.com>

stnie committed:
[None][fix] Add TP information in weight scale loading in WeightOnlyQuantLinearMethod (#7732)

Signed-off-by: Stefan Niebler <82932102+stnie@users.noreply.github.com>

xinhe-nv committed:
[TRTLLM-7250][fix] Add failed cases into waives.txt (#7807)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>

Wanli-Jiang committed:
[TRTLLM-7918][feat] Support kvcache reuse and chunk prefill for phi4mm (#7723)

Signed-off-by: Wanli Jiang <35160485+Wanli-Jiang@users.noreply.github.com>

xinhe-nv committed:
[https://nvbugs/5519462][fix] Add failed cases into waives.txt (#7817)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>
Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

QiJune committed:
[None][chore] remove generated fmha_cubin.h from source tree (#7836)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

byshiue committed:
[None][fix] Revert "Revert "[None][feat] support attention dp for qwen3 dense model"" (#7780)

Signed-off-by: bhsueh <11360707+byshiue@users.noreply.github.com>

MrGeva commented on tensorrt_llm/_torch/auto_deploy/compile/backends/torch_cudagraph.py:
```diff
@@ -86,34 +86,44 @@ def capture_graph(self, *args, **kwargs):
         args, kwargs = self._in_spec.unflatten(self._input_buffers + args_static)
 
-        # capture output once with max batch size to capture output buffers
+        # truncate input buffers to cuda_graph_max_batch_size
+        inputs_truncated = [
+            in_buffer[: self.cuda_graph_max_batch_size] for in_buffer in self._input_buffers
+        ]
+        args, kwargs = self._in_spec.unflatten(inputs_truncated + args_static)
+
+        # capture output once with cuda_graph_max_batch_size to capture output buffers
```
repeat_interleave does not truncate, it only repeats. now that we support cuda_graph_max_batch_size < max_batch_size we need to truncate the buffers to avoid the extra memory footprint when running the next line: out = self.model(*args, **kwargs)

MrGeva replied:
Right, I missed that. thanks for pointing it out! I fixed it now

MrGeva commented on tensorrt_llm/_torch/auto_deploy/compile/backends/torch_cudagraph.py:
```diff
@@ -86,34 +86,44 @@ def capture_graph(self, *args, **kwargs):
         # capture graph now for a range of batch sizes
         for bs in self.cuda_graph_batch_sizes:
             ad_logger.info(f"Capturing graph for batch size: {bs}")
-
-            # setup args, kwargs
-            inputs_truncated = [in_buffer[:bs] for in_buffer in self._input_buffers]
-            args, kwargs = self._in_spec.unflatten(inputs_truncated + args_static)
-
```
I did not remove it, I pulled it up to before the first warm up, to avoid allocating buffers according to the max batch size. the buffers will be allocated according to the cuda_graph_max_batch_size with my change

MrGeva replied:
right, fixing it now

limin2021 committed:
[TRTLLM-6898][feat] Add swapab, tileN64, cga sync support for cute dsl nvfp4 gemm (#7764)

Signed-off-by: Mindy Li <11663212+limin2021@users.noreply.github.com>

MrGeva committed:
removed redundant unflatten

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

dongfengy committed:
[None][doc] Cherry-pick deployment guide update from 1.1.0rc2 branch to main branch (#7774)

Signed-off-by: Dongfeng Yu <dongfengy@nvidia.com>

MrGeva committed:
fixed bug in cuda graph capture loop

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

MrGeva commented:
/bot run

MrGeva commented:
verified the perf is same vs. before my change using:
trtllm-bench --model meta-llama/Llama-3.1-8b throughput --dataset synthetic_128_128.txt --warmup 0 --backend _autodeploy --max_batch_size 256 --extra_llm_api_options /lustre/fs1/portfolios/coreai/users/egeva/autodeploy-dashboard/extra.yaml  --tp=1

extra.yaml:
compile_backend: torch-opt
free_mem_ratio: 0.8
runtime: trtllm
cuda_graph_batch_sizes: [1, 2, 4, 8, 16, 32, 64, 128, 256]
max_batch_size: 256
skip_loading_weights: true

sunnyqgg committed:
[TRTLLM-6746][feat] Enable two-model spec dec for MTP Eagle (#7001)

Signed-off-by: qgai <qgai@nvidia.com>

QiJune committed:
[None][ci] set TORCHINDUCTOR_COMPILE_THREADS correctly (#7800)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

ziyixiong-nv committed:
[https://nvbugs/5522851][fix] Correct the logic to update kv_lens_cuda (#7790)

Signed-off-by: ziyixiong-nv <219238287+ziyixiong-nv@users.noreply.github.com>

yuxianq committed:
[TRTLLM-6994][feat] FP8 Context MLA integration (Cherry-pick https://github.com/NVIDIA/TensorRT-LLM/pull/6059 from release/1.1.0rc2) (#7610)

Signed-off-by: Yuxian Qiu <142763828+yuxianq@users.noreply.github.com>

VALLIS-NERIA committed:
[TRTLLM-6286] [feat] Update CUTLASS to 4.2 and enable SM103 group gemm (#7832)

Signed-off-by: Xiwen Yu <13230610+VALLIS-NERIA@users.noreply.github.com>

chuangz0 committed:
[None][fix] get Local IP by connect remote (#7719)

Signed-off-by: Chuang Zhu <111838961+chuangz0@users.noreply.github.com>

fredricz-20070104 committed:
[TRTLLM-7183][test] Feature fix model issue for disagg serving (#7785)

Signed-off-by: FredricZ-2007 <226039983+fredricz-20070104@users.noreply.github.com>

HuiGao-NV committed:
[https://nvbugs/5481434][feat] cherry-pick fix to reuse pytorch memory segments occupied by cudagraph (#7747)

Signed-off-by: Hui Gao <huig@nvidia.com>

ruodil committed:
[None][test] add deepseek r1/v3 model with chunked prefill cases (#7124)

Signed-off-by: ruodil <200874449+ruodil@users.noreply.github.com>

QiJune committed:
[None][chore] polish error message in cute_dsl_utils.py (#7852)

Signed-off-by: junq <22017000+QiJune@users.noreply.github.com>

lkm2835 committed:
[None][fix] fix load_model_on_cpu on qwen/convert_checkpoint.py (#2382)

Signed-off-by: lkm2835 <lkm2835@gmail.com>
Co-authored-by: Kanghwan <861393+karljang@users.noreply.github.com>

EmmaQiaoCh committed:
[None][infra] Waive failed tests in post-merge (#7859)

Signed-off-by: qqiao <qqiao@nvidia.com>

dominicshanshan committed:
[None][ci] Waive llama3 auto dtype test bug in https://nvbugs/5527956. (#7853)

Signed-off-by: Wangshanshan <30051912+dominicshanshan@users.noreply.github.com>
Signed-off-by: Yanchao Lu <yanchaol@nvidia.com>
Co-authored-by: Yanchao Lu <yanchaol@nvidia.com>

tensorrt-cicd commented:
[PR_Github #19319](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/19319/) [ run ] triggered by Bot

crazydemo committed:
[None][test] Add accuracy benchmark in stress test (#7561)

Signed-off-by: Ivy Zhang <25222398+crazydemo@users.noreply.github.com>

crazydemo committed:
[None][chore] remove cli cases for rtx6k (#7833)

Signed-off-by: Ivy Zhang <25222398+crazydemo@users.noreply.github.com>

lucifer1004 committed:
[None][feat] Support EPLB in Qwen3 MoE (#7443)

Signed-off-by: Gabriel Wu <13583761+lucifer1004@users.noreply.github.com>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

xinhe-nv committed:
[None][chore] Add failed cases into waives.txt (#7841)

Signed-off-by: xinhe-nv <200704525+xinhe-nv@users.noreply.github.com>
Signed-off-by: Xin He (SW-GPU) <200704525+xinhe-nv@users.noreply.github.com>

lancelly committed:
[https://nvbugs/5503440][fix] Fix potential hang due to wrong type of ZMQ socket and protocol for worker_init_status_queue (#7646)

Signed-off-by: Lanyu Liao <lancelly@users.noreply.github.com>

syuoni committed:
[None][doc] Tech blog: Combining Guided Decoding and Speculative Decoding: Making CPU and GPU Cooperate Seamlessly (#7864)

Signed-off-by: Enwei Zhu <21126786+syuoni@users.noreply.github.com>

yuxianq committed:
[https://nvbugs/5522332][fix] Pin numpy version for Gemma. (cherry-pick https://github.com/NVIDIA/TensorRT-LLM/pull/7783) (#7797)

Signed-off-by: Yuxian Qiu <142763828+yuxianq@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #19319](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/19319/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #14506](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/14506/) completed with status: 'SUCCESS'

MatthiasKohl committed:
[TRTLLM-5966][feat] Helix: add custom position ids to MLA kernels (#6904)

Signed-off-by: Matthias Jouanneaux <mjoux@nvidia.com>
Co-authored-by: brb-nv <169953907+brb-nv@users.noreply.github.com>

pcastonguay committed:
[https://nvbugs/5471108][chore] Unwaiving disagg acc test (#7686)

Signed-off-by: Patrice Castonguay <55748270+pcastonguay@users.noreply.github.com>

lucaslie reviewed (approved):
just a few remarks to with regards to the cuda graph logic. Feel free to merge the PR regardless. Thank you!

mikeiovine committed:
[https://nvbugs/5522462][fix] Fix FP8 scout illegal memory access (#7845)

Signed-off-by: Mike Iovine <6158008+mikeiovine@users.noreply.github.com>

karljang committed:
[#7704][chore] Enable MathJax to fix formulas in documentation (#7744)

Signed-off-by: Kanghwan Jang <861393+karljang@users.noreply.github.com>

greg-kwasniewski1 committed:
[TRTLLM-6342][feat] Support for partial sharding from factory (#7393)

Signed-off-by: greg-kwasniewski1 <213329731+greg-kwasniewski1@users.noreply.github.com>
Signed-off-by: Grzegorz Kwasniewski <213329731+greg-kwasniewski1@users.noreply.github.com>

chang-l committed:
[https://nvbugs/5520490][fix] Fix intermittent test failures by avoiding external web data pulls (#7879)

Signed-off-by: Chang Liu (Enterprise Products) <9713593+chang-l@users.noreply.github.com>

syuoni committed:
[None][doc] Update tech blog12 (#7884)

Signed-off-by: Enwei Zhu <21126786+syuoni@users.noreply.github.com>

brb-nv committed:
[TRTLLM-7731][feat] KV cache transmission in disagg with CP on gen side (#7624)

Signed-off-by: Balaram Buddharaju <169953907+brb-nv@users.noreply.github.com>

Superjomn committed:
[TRTLLM-8188][chore] refactor GenerationExecutorWorker with WorkerBase for better code reusing (#7840)

Signed-off-by: Yan Chunwei <328693+Superjomn@users.noreply.github.com>

ziyixiong-nv committed:
[https://nvbugs/5517404][fix] Use the correct cuda graph for dynamic spec dec (#7728)

Signed-off-by: ziyixiong-nv <219238287+ziyixiong-nv@users.noreply.github.com>

VALLIS-NERIA committed:
[TRTLLM-6286] [perf] Add NoSmem epilogue schedule and dynamic cluster shape for sm10x group gemm (#7757)

Signed-off-by: Xiwen Yu <13230610+VALLIS-NERIA@users.noreply.github.com>
Signed-off-by: djns99 <40156487+djns99@users.noreply.github.com>
Co-authored-by: djns99 <40156487+djns99@users.noreply.github.com>

MrGeva committed:
fixed CR comment

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

MrGeva committed:
fixed CR comment

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

MrGeva committed:
Merge branch 'main' into egeva/max_cud_bs

MrGeva committed:
fixed CR

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

MrGeva committed:
fixed CR

Signed-off-by: Eran Geva <19514940+MrGeva@users.noreply.github.com>

tensorrt-cicd commented:
[PR_Github #19439](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/19439/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #19439](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/19439/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #14608](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/14608/) completed with status: 'SUCCESS'

MrGeva commented:
closing this PR because I had used git merge to rebase and it messed up the history and fails on non signed off commits. Opened this new PR instead: https://github.com/NVIDIA/TensorRT-LLM/pull/7888
